File: src/main/java/gargoyle/ct/pref/impl/prop/CTPrefCharProperty.java
package gargoyle.ct.pref.impl.prop;

import gargoyle.ct.pref.CTPreferencesProvider;
import org.jetbrains.annotations.NotNull;

public class CTPrefCharProperty extends CTPrefProperty<Character> {
    @SuppressWarnings("MagicCharacter")
    public CTPrefCharProperty(@NotNull CTPreferencesProvider provider, String name) {
        this(provider, name, '\0');
    }

    public CTPrefCharProperty(@NotNull CTPreferencesProvider provider, String name, Character def) {
        super(Character.class, provider, name, def);
    }
}

File: src/main/kotlin/gargoyle/ct/pref/impl/prop/CTPrefCharProperty.kt
package gargoyle.ct.pref.impl.prop

import gargoyle.ct.pref.CTPreferencesProvider

class CTPrefCharProperty @JvmOverloads constructor(
    provider: CTPreferencesProvider,
    name: String,
    def: Char = '\u0000'
) : CTPrefProperty<Char>(
    Char::class, provider, name, def
)
